pkg/sink/sink.go

```diff
@@ -223,33 +230,57 @@ func (r Sink) ExecuteInterceptors(t *triggersv1.EventListenerTrigger, in *http.R
 		case i.GitLab != nil:
 			interceptor = gitlab.NewInterceptor(i.GitLab, r.KubeClientSet, r.EventListenerNamespace, log)
 		case i.CEL != nil:
-			interceptor = cel.NewInterceptor(i.CEL, r.KubeClientSet, r.EventListenerNamespace, log)
+			interceptor = cel.NewInterceptor(r.KubeClientSet, log)
 		case i.Bitbucket != nil:
 			interceptor = bitbucket.NewInterceptor(i.Bitbucket, r.KubeClientSet, r.EventListenerNamespace, log)
 		default:
-			return nil, nil, fmt.Errorf("unknown interceptor type: %v", i)
-		}
-		var err error
-		resp, err = interceptor.ExecuteTrigger(request)
-		if err != nil {
-			return nil, nil, err
+			return nil, nil, nil, fmt.Errorf("unknown interceptor type: %v", i)
 		}
 
-		// Set the next request to be the output of the last response to enable
-		// request chaining.
-		request = &http.Request{
-			Method: http.MethodPost,
-			Header: resp.Header,
-			URL:    in.URL,
-			Body:   ioutil.NopCloser(resp.Body),
+		// Immutable Interface check
+		if interceptorInterface, ok := interceptor.(triggersv1.InterceptorInterface); ok {
+			// Set per interceptor config params to the request
+			request.InterceptorParams = interceptors.GetInterceptorParams(i)
+			interceptorResponse = interceptorInterface.Process(context.Background(), &request)
```

Comment:
We should add the context as a parameter so that we can accurately pass through timeouts and tracing information (if any).

Reply:
ack. the only reason I did not do it here was because there was no existing context to plumb through. Was planning on adding it a follow up PR where we move the interceptors to run in a separate HTTP service. What do you think?

Reply:
sgtm